### PR TITLE
Expose the configuration options of auth state

### PR DESCRIPTION
**Pull request by tothandras**

### CODE REVIEW

File: jupyterhub/templates/hub/deployment.yaml [hunk 1]
```diff
@@ -105,6 +105,13 @@ spec:
         - name: {{ $key | quote }}
           value: {{ $value | quote }}
         {{ end -}}
+        {{ if .Values.auth.state.enable }}
```
Comment (yuvipanda) [inline jupyterhub/templates/hub/deployment.yaml]: This should probably be put in the hub secret and then put here via a ref I think.

File: jupyterhub/values.yaml [hunk 2]
```diff
@@ -102,6 +102,9 @@ auth:
     users:
   dummy:
     password:
+  state:
+    enable: false
```
Comment (yuvipanda) [inline jupyterhub/values.yaml]: we use 'enabled' elsewhere. Use that here too to be consistent?

**Comment by yuvipanda** [comment]
This is awesome, thanks for the patch! I'd like to hold off merging this until we get 0.5 out the door (today or tomorrow), hope that is ok!

File: jupyterhub/templates/hub/deployment.yaml (hunk 1 above)
Comment (tothandras) [reply jupyterhub/templates/hub/deployment.yaml]: I think I already do that. Can you clarify?

File: jupyterhub/values.yaml (hunk 2 above)
Comment (tothandras) [reply jupyterhub/values.yaml]: Okay!

File: jupyterhub/templates/hub/deployment.yaml (hunk 1 above)
Comment (yuvipanda) [reply jupyterhub/templates/hub/deployment.yaml]: Nope, I just hadn't fully woken up yet. Aaa, sorryy!!

File: jupyterhub/values.yaml (hunk 2 above)
Comment (tothandras) [reply jupyterhub/values.yaml]: Updated! 🙂

**Comment by tothandras** [comment]
@yuvipanda Any updates on when it might be?

**Comment by yuvipanda** [comment]
Apologies for the delay! I've merged this in now!

**Pull request merged**